From 5dc329d21d710ac71778756d8ca9cde962e12331 Mon Sep 17 00:00:00 2001
From: David Sterba <dsterba@suse.cz>
Date: Tue, 2 Sep 2014 12:03:03 +0200
Subject: [PATCH 90/90] btrfs; disable defrag of tree roots

Recursive defrag is supported by userspace tools, defrag of a dir that
in fact defrags tree roots is wrong and has to be fixed in another way.

Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/ioctl.c | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/fs/btrfs/ioctl.c b/fs/btrfs/ioctl.c
index 0c7556a..e1dd639 100644
--- a/fs/btrfs/ioctl.c
+++ b/fs/btrfs/ioctl.c
@@ -2599,15 +2599,8 @@ static int btrfs_ioctl_defrag(struct file *file, void __user *argp)
 
 	switch (inode->i_mode & S_IFMT) {
 	case S_IFDIR:
-		if (!capable(CAP_SYS_ADMIN)) {
-			ret = -EPERM;
-			goto out;
-		}
-		ret = btrfs_defrag_root(root);
-		if (ret)
-			goto out;
-		ret = btrfs_defrag_root(root->fs_info->extent_root);
-		break;
+		ret = 0;
+		goto out;
 	case S_IFREG:
 		if (!(file->f_mode & FMODE_WRITE)) {
 			ret = -EINVAL;
-- 
1.9.1

